src/Microsoft.VisualStudio.ProjectSystem.Managed/Threading/Tasks/TaskDelayScheduler.cs

```diff
@@ -43,13 +54,16 @@ public JoinableTask ScheduleAsyncTask(Func<CancellationToken, Task> operation, C
                     return;
                 }
 
-                try
+                if (!immediate)
                 {
-                    await Task.Delay(_taskDelayTime, nextToken);
-                }
-                catch (OperationCanceledException)
-                {
-                    return;
+                    try
+                    {
+                        await Task.Delay(_taskDelayTime, nextToken);
+                    }
+                    catch (OperationCanceledException)
+                    {
+                        return;
+                    }
                 }
 
                 if (!nextToken.IsCancellationRequested)
```

Comment:
Not a biggie but it might read better if this check was moved inside your new `if` block. You only need to check the token a second time if `immediate` is `false`.